server/server.go

```diff
@@ -126,26 +122,31 @@ func (cs *cribbageServer) ginPostCreatePlayer(c *gin.Context) {
 }
 
 func (cs *cribbageServer) ginPostCreateInteraction(c *gin.Context) {
-	pID := getPlayerID(c, `playerId`)
+	var cir network.CreateInteractionRequest
+	err := c.ShouldBindJSON(&cir)
+	if err != nil {
+		c.String(http.StatusInternalServerError, `Error: %s`, err)
+		return
+	}
+	pID := cir.PlayerID
 	if pID == model.InvalidPlayerID {
 		c.String(http.StatusBadRequest, `Needs playerId`)
 		return
 	}
 
-	var mode interaction.Mode
-	switch c.Param(`mode`) {
-	case `localhost`:
-		mode = interaction.Localhost
+	var pm interaction.PlayerMeans
+	switch {
+	case cir.LocalhostPort != network.EmptyInteraction:
```

Comment:
💡 I think this could be simplified to instead be `len(cir.LocalhostPort) > 0`. comparing against `network.EmptyInteraction` is confusing...